packages/web3-eth-contract/src/utils.ts

```diff
@@ -191,16 +214,13 @@ export const getCreateAccessListParams = ({
 			input: contractOptions.input,
 			maxPriorityFeePerGas: contractOptions.maxPriorityFeePerGas,
 			maxFeePerGas: contractOptions.maxFeePerGas,
+			data: contractOptions.data,
 		},
 		options as unknown as Record<string, unknown>,
 	) as unknown as TransactionForAccessList;
 
-	if (!txParams.input || abi.type === 'constructor') {
-		txParams = {
-			...txParams,
-			input: encodeMethodABI(abi, params, txParams.input as HexString),
-		};
-	}
+	const dataInput = dataInputEncodeMethodHelper(txParams, abi, params, options?.dataInputFill);
+	txParams = { ...txParams, data: dataInput.data, input: dataInput.input };
```

Comment:
This means when data or input are not supposed to be filled we will have in json: data: undefined or input: undefined, right? I think this might cause some providers to throw when we provide for example: data: '0x...', input: undefined. Or this would not be the case?

Reply:
will always have data or input filled in [dataInputEncodeMethodHelper](https://github.com/web3/web3.js/pull/6377/files/0df6e8b05ecc272448deef313ede102a5c84bca4#diff-d3f00787c6c9b9f20005a07bdad48461cb7370797a0391442e7025413204d743R48) is run (will default to input)